plugin-server/src/main/runner/on-event.ts

```diff
@@ -0,0 +1,25 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { runInstrumentedFunction } from '../../main/ingestion-queues/ingest-event'
+import { Hub, RunnerWorkerMethods } from '../../types'
+
+export async function onEvent(
+    server: Hub,
+    workerMethods: RunnerWorkerMethods,
+    event: PluginEvent,
+    checkAndPause?: () => void // pause incoming messages if we are slow in getting them out again
+) {
+    const isSnapshot = event.event === '$snapshot'
+
+    const processedEvent: PluginEvent | null = event
+
+    checkAndPause?.()
+
+    await runInstrumentedFunction({
+        server,
+        event: processedEvent,
```

Comment:
this needs to be the processed event, i.e. the event with geoIP etc stuff, but here we seem to be using just the normal event? Furthermore drop `const processedEvent: PluginEvent | null = event` and make the input param be the `processedEvent` see my comment in plugin-server/src/worker/ingestion/ingest-event.ts

Reply:
Yeah this was a typo from copy 🍝 . Notice how #9380 does this now